arangod/Aql/QuerySnippet.cpp

```diff
@@ -474,23 +530,36 @@ ResultT<std::unordered_map<ExecutionNode*, std::set<ShardID>>> QuerySnippet::pre
       auto collectionAccessingNode = dynamic_cast<CollectionAccessingNode*>(exp.node);
       TRI_ASSERT(collectionAccessingNode != nullptr);
       collectionAccessingNode->setUsedShard(*myExp.begin());
-      if (exp.doExpand) {
-        TRI_ASSERT(!collectionAccessingNode->isUsedAsSatellite());
-        // All parts need to have exact same size, they need to be permutated pairwise!
-        TRI_ASSERT(numberOfShardsToPermutate == 0 || myExp.size() == numberOfShardsToPermutate);
-        // set the max loop index (note this will essentially be done only once)
-        numberOfShardsToPermutate = myExp.size();
-        if (numberOfShardsToPermutate > 1) {
-          // Only in this case we really need to do an expansion
-          // Otherwise we get away with only using the main stream for
-          // this server
-          // NOTE: This might differ between servers.
-          // One server might require an expansion (many shards) while another does not (only one shard).
-          localExpansions.emplace(exp.node, std::move(myExp));
-        }
-      } else {
-        TRI_ASSERT(myExp.size() == 1);
+    }
```

Comment:
I am not sure, but it looks like the scope here is somewhat off.

Reply:
Scope is :gut: here :)